src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -53,7 +66,363 @@ class DataFrameWriter:
+                asynchronously and returns an :class:`AsyncJob`.
+            **kwargs: Additional options to be passed for format-specific or copy-specific configurations.
+
+        Notes:
+            - If the `format` parameter is specified, it will take precedence over the format set through the `format` method or previous `option` calls.
+
+        Alternatively, you can use the `format` method and specify the format, and then use the `save` method to specify the path:
+            df = spark.read.format("csv").option("header", "true").save("path/to/your/file.csv")
+
+        The following example demonstrate how to use a DataFrameWriter.
+            >>> # Create a temp stage to run the example code.
+            >>> _ = session.sql("CREATE or REPLACE temp STAGE mystage").collect()
+
+        Example 1:
+            Loading the first two columns of a CSV file and skipping the first header line:
+            >>> from snowflake.snowpark.types import StructType, StructField, IntegerType, StringType
+            >>> _ = session.sql("create file format if not exists csv_format type=csv skip_header=1 null_if='none';").collect()
+            >>> _ = session.file.put("tests/resources/testCSVspecialFormat.csv", "@mystage", auto_compress=False)
+            >>> # Define the schema for the data in the CSV files.
+            >>> schema = StructType([StructField("ID", IntegerType()),StructField("USERNAME", StringType()),StructField("FIRSTNAME", StringType()),StructField("LASTNAME", StringType())])
+            >>> # Create a DataFrame that is configured to load data from the CSV files in the stage with a given schema and format.
+            >>> df = session.read.load("@mystage/testCSVspecialFormat.csv",format="csv",schema=schema,format_name="csv_format")
+            >>> # Load the data into the DataFrame and return an array of rows containing the results.
+            >>> df.collect()
+            [Row(ID=0, USERNAME='admin', FIRSTNAME=None, LASTNAME=None), Row(ID=1, USERNAME='test_user', FIRSTNAME='test', LASTNAME='user')]
+        """
+        if not path and self._path:
+            path = self._path
+        if format:
+            self.format(format)
+        if partitionBy:
+            self.partitionBy(partitionBy)
+        for key, value in kwargs.items():
+            self.option(key, value)
+        if mode:
+            self.mode(mode)
+        if not self._file_format_type:
+            raise SnowparkClientExceptionMessages.DF_MUST_PROVIDE_FILETYPE_FOR_WRITING_FILE()
+        return self._write_to_location(
+            path, self._file_format_type, self._save_mode.value, block
+        )
+
+    def partitionBy(self, partition_by: Optional[ColumnOrSqlExpr]) -> "DataFrameWriter":
+        """Specifies an expression used to partition the unloaded table rows into separate files. It can be a :class:`Column`, a column name, or a SQL expression."""
```

Comment:
```suggestion
    def partition_by(self, partition_by: Optional[ColumnOrSqlExpr]) -> "DataFrameWriter":
        """Specifies an expression used to partition the unloaded table rows into separate files. It can be a :class:`Column`, a column name, or a SQL expression."""
```
we generally don't use CamelCase in this repository

Reply:
noted

Reply:
adding an alias is fine if we need to make this compatible for snowconvert